### PR TITLE
add model specs, add model validations

### PR DESCRIPTION
**Pull request by zmprunella**

### CODE REVIEW

File: spec/models/cat_spec.rb [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+      cat = Cat.create age: 3, enjoys: "all the attention"
+      expect(cat.errors[:name]).to_not be_empty
+  end
+  it "should have an age" do
+    cat = Cat.create name: 'Toasty', enjoys: "all the attention"
+    expect(cat.errors[:age]).to_not be_empty
+  end
+  it "should have an enjoys" do
+    cat = Cat.create name: 'Toasty', age: 2
+    expect(cat.errors[:enjoys]).to_not be_empty
```
Comment (WalkerTexas-coder) [inline spec/models/cat_spec.rb]: Let's add the spec for image and for enjoys length

File: spec/requests/cats_spec.rb [hunk 2]
```diff
@@ -29,4 +29,26 @@
       expect(cat.enjoys).to eq 'Looking at trees'
     end
   end
+
+  describe "meaningful status codes" do
```
Comment (WalkerTexas-coder) [inline spec/requests/cats_spec.rb]: Yea?

File: spec/requests/cats_spec.rb [hunk 3]
```diff
@@ -29,4 +29,26 @@
...
+        cat: {
+          age:2,
+          enjoys: 'long walks on the beach, sunbathing and eating tuna out of a can.'
+        }
+      }
+      post '/cats', params: cat_params
+      expect(response.status).to eq 422
+      cat = JSON.parse(response.body)
+      expect(cat['name']).to include "can't be blank"
+    end
```
Comment (WalkerTexas-coder) [inline spec/requests/cats_spec.rb]: Missing 3 other request specs

File: spec/requests/cats_spec.rb [hunk 4]
```diff
@@ -29,4 +29,26 @@
...
+
+    it "enjoys must have minimum length of ten" do
+      cat = Cat.create name:'Moose', age:4,enjoys:'hiking', image:'https://m.media-amazon.com/images/I/812nB8o+guL._AC_SX466_.jpg'
+
+      expect(cat.errors[:enjoys]).to_not be_empty
+    end
```
Comment (WalkerTexas-coder) [inline spec/requests/cats_spec.rb]: Missing the request aspect of this spec. This looks like it belongs in the models/spec

**Review by WalkerTexas-coder — DISMISSED** [review]
Check out my feedback!

File: spec/requests/cats_spec.rb [hunk 5]
```diff
@@ -29,4 +29,25 @@
       expect(cat.enjoys).to eq 'Looking at trees'
     end
   end
+
+  describe "meaningful status codes" do
```
Comment (elysemontano) [inline spec/requests/cats_spec.rb]: Would recommend a more descriptive name for your test here.  More specifically than status codes, what is your intent with these tests?

**Review by elysemontano — CHANGES_REQUESTED** [review]
You should be checking validations on each attribute in your request spec as well.  Think about what you are testing against.  When testing validations, you will need to consider what error code you get back if you try to create a cat without that attribute.

File: spec/models/cat_spec.rb [hunk 6]
```diff
@@ -1,5 +1,31 @@
...
+
+      expect(cat.errors[:age]).to_not be_empty
+    end
+   it "should have an enjoys" do
+      cat = Cat.create name: 'Toasty', age: 2, image: 'https://m.media-amazon.com/images/I/812nB8o+guL._AC_SX466_.jpg'
+
+      expect(cat.errors[:enjoys]).to_not be_empty
+    end
+   it "should have an image" do
+      cat = Cat.create name: 'Toasty', age: 2
```
Comment (elysemontano) [inline spec/models/cat_spec.rb]: This cat should enjoy something.

File: spec/requests/cats_spec.rb [hunk 7]
```diff
@@ -3,30 +3,52 @@
...
-      expect(cat.age).to eq 5
-      expect(cat.enjoys).to eq 'Looking at trees'
+      expect(cat.name).to eq('Giraffe')
+      expect(cat.age).to eq(3)
+      expect(cat.enjoys).to eq('Eating tuna')
+      expect(cat.image).to eq('https://townsquare.media/site/396/files/2021/06/attachment-Cat-Walking-On-Beach.jpg')
+    end
+  end
+
+  describe "can request validation" do
```
Comment (elysemontano) [inline spec/requests/cats_spec.rb:38]: Much better naming convention!  I don't see request validations for age, enjoys, and image.

**Review by elysemontano — CHANGES_REQUESTED** [review]
Almost there, left a few more comments.

**Review by elysemontano — APPROVED** [review]
Looks good!

**Pull request merged**